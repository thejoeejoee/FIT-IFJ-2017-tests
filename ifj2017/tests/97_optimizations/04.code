' constant propagation into loop II

scope
	dim counter as integer = 50
	dim constant as integer = 8 + 7

	do while counter > 0
		print (10.2 / 98 + (counter \ 3 * 6.85) - 14) / (counter + 5) + (constant + 9 - 8) * ((constant+17.21)\(10.8 * counter));
		counter = counter - 1
		print !"\n";
	loop

end scope