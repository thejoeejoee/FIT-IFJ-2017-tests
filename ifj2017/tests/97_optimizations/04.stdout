-0.216289
-0.220295
-0.224451
-0.228768
-0.233253
-0.237918
-0.242774
-0.247832
-0.253105
-0.280346
-0.286576
-0.293089
-0.299905
-0.307046
-0.314535
-0.322398
-0.330665
-0.339366
-0.348538
-0.35822
-0.368455
-0.379292
-0.390785
-0.402997
-0.415997
-0.429864
-0.444687
-0.460569
-0.477627
-0.495997
-0.555837
-0.578997
-0.60417
-0.631633
-0.66171
-0.694796
-0.731364
-0.771995
-0.817407
-0.868495
-0.926395
-0.992566
-1.06892
-1.15799
-1.26327
-1.38959
-1.54399
 14.263
 14.0149
 29.684
